### PR TITLE
feat(auth): token-regex hardening with negative-test fixture (T3.B)

### PR DESCRIPTION
**Pull request by teng-lin**
## Summary
- Unify token extraction (SNlM0e, FdrFJe, arbitrary WIZ_global_data fields) behind a single ``extract_wiz_field`` helper in ``auth.py``.
- Add typed ``AuthExtractionError(RPCError)`` carrying a sanitized 200-char HTML preview so schema drift surfaces a diagnostic instead of silently returning garbage.
- Cover drift, variants, and preview behavior with fixture-backed tests.

Parent plan: ``.sisyphus/plans/tier-3-implementation.md`` (PR-T3.B).

## Sites consolidated

Enumerated via ``grep -rn 're\.search' src/notebooklm/auth.py src/notebooklm/client.py``:

| File | Symbol | Field |
|---|---|---|
| ``src/notebooklm/auth.py:826`` | ``extract_csrf_from_html`` | ``SNlM0e`` |
| ``src/notebooklm/auth.py:858`` | ``extract_session_id_from_html`` | ``FdrFJe`` |
| ``src/notebooklm/client.py:245`` | ``NotebookLMClient.refresh_auth`` | ``SNlM0e`` |
| ``src/notebooklm/client.py:254`` | ``NotebookLMClient.refresh_auth`` | ``FdrFJe`` |

All four now route through ``extract_wiz_field`` and inherit the new variant-tolerant matching matrix.

## Helper contract

```python
def extract_wiz_field(html: str, key: str, *, strict: bool = True) -> str | None
```

Patterns tried in priority order:
1. Canonical double-quoted: ``"key":"value"`` (with optional whitespace around the colon)
2. Single-quoted: ``'key':'value'``
3. HTML-escaped: ``&quot;key&quot;:&quot;value&quot;``

Empty values pass through verbatim (``"SNlM0e":""`` returns ``""``, not ``None``) because some endpoints legitimately emit empty tokens.

On drift with ``strict=True``: raises ``AuthExtractionError`` carrying ``key`` and a 200-char ``payload_preview`` (whitespace-collapsed for readability). On ``strict=False``: returns ``None`` so callers can fall back gracefully.

## Backward compatibility

The legacy ``extract_csrf_from_html`` / ``extract_session_id_from_html`` helpers and ``refresh_auth`` continue raising ``ValueError`` with the historical message substrings (``"CSRF token not found"``, ``"Failed to extract session ID"``, etc.) so existing ``except ValueError`` handlers and tests are unaffected. Only the underlying matching machinery changed.

## New fixture + tests

- ``tests/fixtures/notebooklm_html.html`` — sanitized real-shape NotebookLM homepage capture. Every secret replaced with a ``SCRUBBED_*`` sentinel; structural shape mirrors what ``notebooklm.google.com`` actually emits.
- ``tests/unit/test_token_regex.py`` — 18 tests covering:
  - Fixture happy paths for ``SNlM0e``, ``FdrFJe``, and ``oM1Kwf`` (account email).
  - Variant matrix: single-quoted, HTML-escaped, whitespace tolerance.
  - Empty-value passthrough.
  - Canonical pattern wins when multiple variants coexist.
  - Strict / non-strict modes (default is strict).
  - Preview truncation, whitespace collapse, inclusion in ``__str__``.
  - ``AuthExtractionError`` is an ``RPCError`` subclass.
  - Defensive key escaping (field names with regex metacharacters).

## Test plan
- [x] ``uv run ruff format .``
- [x] ``uv run ruff check .``
- [x] ``uv run mypy src/notebooklm --ignore-missing-imports`` (clean)
- [x] ``uv run pytest`` (3048 passed, 12 skipped)
- [x] All 18 new tests in ``tests/unit/test_token_regex.py`` pass
- [x] All 362 pre-existing auth/client/keepalive/refresh-race tests still pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Publicly expose AuthExtractionError and add a helper to extract auth fields from page HTML.

* **Bug Fixes**
  * More robust token extraction across varied HTML formats; preserves legacy human-readable error messages on auth failures.

* **Tests**
  * Added comprehensive tests and an HTML fixture covering extraction variants, diagnostics, and edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/teng-lin/notebooklm-py/pull/463)

### CODE REVIEW

**Comment by teng-lin** [comment]
@claude review

File: src/notebooklm/auth.py [hunk 1]
```diff
@@ -805,6 +806,71 @@ def extract_cookies_from_storage(storage_state: dict[str, Any]) -> dict[str, str
...
+    ``extract_csrf_from_html`` regex so we don't regress.
+    """
+    escaped = re.escape(key)
+    return [
+        # 1. Canonical double-quoted: "key":"value"  (or  "key" : "value")
+        re.compile(rf'"{escaped}"\s*:\s*"([^"]*)"'),
+        # 2. Single-quoted variant: 'key':'value'
+        re.compile(rf"'{escaped}'\s*:\s*'([^']*)'"),
+        # 3. HTML-escaped: &quot;key&quot;:&quot;value&quot;
+        re.compile(rf"&quot;{escaped}&quot;\s*:\s*&quot;([^&]*)&quot;"),
```
[the remark this answers is not in the document]
Comment (teng-lin) [reply src/notebooklm/auth.py]: Addressed in commit c7b39b4: _build_wiz_field_patterns now uses the standard "string with escapes" idiom ([^"\\]*(?:\\.[^"\\]*)*) for the double- and single-quoted variants, and the HTML-escaped variant switched to a tempered-dot lookahead so embedded entities like &amp; survive. Three new tests pin the escape handling.

File: src/notebooklm/exceptions.py [hunk 2]
```diff
@@ -276,6 +278,52 @@ class AuthError(RPCError):
...
+        key: str,
+        payload_preview: str,
+        *,
+        message: str | None = None,
+    ):
+        self.key = key
+        # Collapse runs of whitespace so the preview stays compact and useful
+        # even when the upstream HTML is heavily indented or contains newlines.
+        collapsed = re.sub(r"\s+", " ", payload_preview).strip()
+        self.payload_preview = collapsed[: self.PREVIEW_LENGTH]
```
[the remark this answers is not in the document]
Comment (teng-lin) [reply src/notebooklm/exceptions.py]: Addressed in commit c7b39b4: AuthExtractionError.__init__ now slices to PREVIEW_LENGTH * 5 before running the whitespace-collapse regex. The 5x headroom guarantees enough non-whitespace characters survive even on heavily-indented HTML. A new test_preview_slices_before_regex_substitution exercises a 5MB payload to confirm the failure path no longer pays the full-body substitution cost.

**Comment by teng-lin** [comment]
Addressed in 5fb5efa: removed the no-op `__str__` override on `AuthExtractionError` — `super().__str__()` resolves to `Exception.__str__` via inheritance since `RPCError` doesn't override it. Thanks for the catch!
